### PR TITLE
Start implementing vehicle update.

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Label all object array addresses

### CODE REVIEW

File: src/openloco/objects/objectmgr.cpp [hunk 1]
```diff
@@ -5,7 +5,43 @@ using namespace openloco::interop;
...
+    loco_global_array<track_object *, 1, 0x0050C914> _trackObjects;
+    loco_global_array<road_station_object *, 1, 0x0050C934> _roadStationObjects;
+    loco_global_array<road_extra_object *, 1, 0x0050C974> _roadExtraObjects;
+    loco_global_array<road_object *, 1, 0x0050C984> _roadObjects;
+    loco_global_array<airport_object *, 1, 0x0050C9A4> _airportObjects;
+    loco_global_array<dock_object *, 1, 0x0050C9C4> _dockObjects;
+    loco_global_array<vehicle_object *, 1, 0x0050C9E4> _vehicleObjects;
+    loco_global_array<tree_object *, 1, 0x0050CD64> _treeObjects;
+    loco_global_array<snow_object *, 1, 0x0050CE64> _snowObjects;
+    loco_global_array<climate_object *, 1, 0x0050CE68> _cliamteObjects;
```
Comment (marijnvdwerf) [inline src/openloco/objects/objectmgr.cpp]: typo

File: src/openloco/objects/objectmgr.cpp [hunk 2]
```diff
@@ -5,7 +5,43 @@ using namespace openloco::interop;
...
+    loco_global_array<currency_object *, 1, 0x0050C5D4> _currencyObjects;
+    loco_global_array<steam_object *, 1, 0x0050C5D8> _steamObjects;
+    loco_global_array<rocks_object *, 1, 0x0050C658> _rockObjects;
+    loco_global_array<water_object *, 1, 0x0050C678> _waterObjects;
+    loco_global_array<surface_object *, 1, 0x0050C67C> _surfaceObjects;
+    loco_global_array<town_name_object *, 1, 0x0050C6FC> _townNameObjects;
     loco_global_array<cargo_object *, 32, 0x0050C700> _cargoObjects;
+    loco_global_array<wall_object *, 1, 0x0050C780> _wallObjects;
+    loco_global_array<train_signal_object *, 1, 0x0050C800> _trainSignalObjects;
+    loco_global_array<level_crossing_object *, 1, 0x0050C840> _levlCrossingObjects;
```
Comment (marijnvdwerf) [inline src/openloco/objects/objectmgr.cpp]: typo in name

**Comment by IntelOrca** [comment]
Object counts are now in `master`.

File: src/openloco/viewportmgr.h [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include "window.h"
+
+namespace openloco::ui::viewportmgr
+{
+    viewport * begin();
+    viewport * end();
+}
```
Comment (janisozaur) [inline src/openloco/viewportmgr.h]: nom nom

File: src/openloco/things/vehicle.cpp [hunk 4]
```diff
@@ -243,3 +249,60 @@ void openloco::vehicle::sub_4AAC4E()
         break;
     }
 }
+
+// 0x4CBB01
+void openloco::vehicle::invalidate_sprite()
+{
+    if (sprite_left == 0x8000)
```
Comment (janisozaur) [inline src/openloco/things/vehicle.cpp]: > `error: comparison of constant 32768 with expression of type 'int16_t' (aka 'short') is always false`

File: src/openloco/objects/objectmgr.cpp [hunk 5]
```diff
@@ -5,7 +5,43 @@ using namespace openloco::interop;
...
+    loco_global_array<object_repository_item, 64, 0x4FE0B8> object_repository;
+
+    loco_global_array<interface_skin_object *, 1, 0x0050C3D0> _interfaceObjects;
+    loco_global_array<sound_object *, 128, 0x0050C3D4> _soundObjects;
+    loco_global_array<currency_object *, 1, 0x0050C5D4> _currencyObjects;
+    loco_global_array<steam_object *, 32, 0x0050C5D8> _steamObjects;
+    loco_global_array<rock_object *, 8, 0x0050C658> _rockObjects;
+    loco_global_array<water_object *, 1, 0x0050C678> _waterObjects;
+    loco_global_array<land_object *, 32, 0x0050C67C> _landObjects;
+    loco_global_array<town_name_object *, 1, 0x0050C6FC> _townNameObjects;
```
Comment (duncanspumpkin) [inline src/openloco/objects/objectmgr.cpp]: Town names

File: src/openloco/objects/vehicle_object.h [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        uint8_t pad_08[0x0E - 0x08];
+        uint8_t sprite_num;     // 0x0E
+        uint8_t pad_0F[0x1E - 0xF];
+    };
+
+    struct vehicle_object
+    {
+        string_id name; // 0x00 probably not confirmed
+        uint8_t vehicle_class;  // 0x02
+        uint8_t type;           // 0x03
```
Comment (duncanspumpkin) [inline src/openloco/objects/vehicle_object.h]: type (controls in which tab the vehicle is when you want to buy it (0=trains,1=busses,2=trucks,3=trams,4=planes,5=ships))

File: src/openloco/objects/vehicle_object.h [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        vehicle_object_unk var_24[4];
+        vehicle_object_sprite sprites[4];   // 0x3C
+        uint8_t pad_B4[0xD8 - 0xB4];
+        uint16_t power;         // 0xD8
+        uint16_t speed;         // 0xDA
+        uint16_t rack_speed;    // 0xDC
+        uint16_t weight;        // 0xDE
+        uint16_t flags;         // 0xE0
+        uint8_t pad_E1[0x10E - 0xE1];
+        uint8_t vis_fx_height;  // 0x10E
```
Comment (duncanspumpkin) [inline src/openloco/objects/vehicle_object.h]: (height of the smoke or steam effect)

**Pull request merged**